### DISCUSSION TITLE
Make an image greyscale but one color (color-spotting)

### DISCUSSION THREAD

**Discussion by lennbarkey** [ImageMagick/ImageMagick#4404] [Help] [answered]
Hello there.👋🏻

I am relatively new to IM and want to convert an image to greyscale but exclude one color (and adjacend ones for that matter), this is known as _"color-spotting"_ in Photoshop, for example. I've found this old thread from the IM legacy forums: [https://legacy.imagemagick.org/discourse-server/viewtopic.php?t=7053](url). In it, someone posted a small script, which i converted to the windows syntax (i am on ImageMagick-7.1.0-portable-Q16-x86) as follows:

`magick convert input.jpg -alpha Set ^`
`( +clone  -fuzz 30% -transparent blue  ) ^`
`-compose DstOut -composite output.jpg`

When i ran this the output image is an all black image, only 1 color according to IrfanView. What is wrong with this script, my conversion of the syntax or is there another way to achieve the "color-spotting"-effect nowadays?

**Comment by snibgo** [comment, marked as answer]
I'm not sure what you want. Perhaps you start with a colour image. Pixels that are close to a certain colour should remain as they are, and other pixels should be turned into their shade of gray.

Windows BAT script:
```
magick ^
  rose: -alpha opaque ^
  ( -clone 0 -colorspace Gray ) ^
  ( -clone 0 -fuzz 30%% -transparent red ) ^
  -compose Over -composite ^
  x.png
```
![x](https://user-images.githubusercontent.com/33812368/138557439-f505f228-1dc3-441d-9485-b4d88ecf02ac.png)

EDIT: Removed superfluous "+write x0.png" from second clone.

**Reply by lennbarkey** [reply]
This is exactly what i wanted to do. Thank you!

**Comment by snibgo** [comment]
Oops, I left in "+write x0.png" which may be useful for debugging, but you can remove it.
